tools/kore-proof-trace-test/main.cpp

```diff
@@ -0,0 +1,68 @@
+#include <kllvm/binary/ProofTraceParser.h>
+
+#include <llvm/Support/CommandLine.h>
+
+#include <string>
+
+using namespace llvm;
+using namespace kllvm;
+
+cl::OptionCategory KoreProofTraceTestCat("kore-proof-trace options");
+
+cl::opt<std::string> InputFilename(
+    cl::Positional, cl::desc("<input file>"), cl::Required,
+    cl::cat(KoreProofTraceTestCat));
+
+int main(int argc, char **argv) {
+  cl::HideUnrelatedOptions({&KoreProofTraceTestCat});
+  cl::ParseCommandLineOptions(argc, argv);
+
+  ProofTraceParser Parser(false, 3u);
+  LLVMRewriteTrace Trace;
+  if (!Parser.parse_proof_trace(InputFilename, Trace)) {
+    return 1;
+  }
+
+  // check that there is a initial configuration
+  if (!(Trace.initialConfig.isPattern()
+        && Trace.initialConfig.getKOREPattern())) {
+    return 1;
+  }
+
+  // check that the trace after the initial configuration is 4 events long
+  if (Trace.trace.size() != 4u) {
+    return 1;
+  }
+
+  // check that the first event is the rewrite a() => b()
+  const auto *Rule1 = dynamic_cast<LLVMRuleEvent const *>(
+      Trace.trace[0].getStepEvent().get());
```

Comment:
Prefer `std::dynamic_pointer_cast` rather than `dynamic_cast` on the contained pointer